src/resolution/instantiation.ts

```diff
@@ -11,35 +11,28 @@ interface InstanceCreationInstruction{
     propertyRequests:interfaces.Request[]
 }
 
-type CreateInstanceWithInjectionArg<T> = InstanceCreationInstruction & {constr: interfaces.Newable<T>,}
+type ResolvedRequests = InstanceCreationInstruction & {isAsync:boolean}
```

Comment:
I think we can express this as an interface
```ts
interface ResolvedRequests extends InstanceCreationInstruction { 
    isAsync:boolean;
}
```

Reply:
Sure if you prefer